src/ustreamer/device.c

```diff
@@ -319,8 +320,14 @@ int us_device_grab_buffer(us_device_s *dev, us_hw_buffer_s **hw) {
 
 	do {
 		struct v4l2_buffer new = {0};
-		new.type = V4L2_BUF_TYPE_VIDEO_CAPTURE;
+		new.type = dev->capture_type;
 		new.memory = dev->io_method;
+		if (dev->capture_type == V4L2_BUF_TYPE_VIDEO_CAPTURE_MPLANE) {
+			new.length = 1;
```

Comment:
from the offical document, the length here should be `VIDEO_MAX_PLANES`

> For the multi-planar API the application sets this to the number of elements in the planes array. The driver will fill in the actual number of valid elements in that array.


ref:
https://www.kernel.org/doc/html/v4.9/media/uapi/v4l/buffer.html#c.v4l2_buffer